src/test/ui/issues/issue-12127.nll.stderr

```diff
@@ -0,0 +1,18 @@
+error[E0382]: use of moved value: `f`
+  --> $DIR/issue-12127.rs:21:9
+   |
+LL |         f();
+   |         - value moved here
+LL |         f();
+   |         ^ value used here after move
+   |
+note: closure cannot be invoked more than once because it moves the variable `x` out of its environment
+  --> $DIR/issue-12127.rs:18:39
+   |
+LL |     let f = to_fn_once(move|| do_it(&*x));
+   |                                       ^
```

Comment:
This test doesn't have this note in the AST borrow checker. A brief glance made me think it was right but I'm not too sure.

Reply:
The closure only implements `FnOnce` because of `to_fn_once`, otherwise it would be a `Fn` closure (it doesn't drop/move `x` when called).